docs/architecture/maui/mvvm.md

```diff
@@ -0,0 +1,364 @@
+
+> [!TIP]
+> Keep view models and views independent
+
+The binding of views to a property in a data source should be the view's principal dependency on its corresponding view model. Specifically, don't reference view types, such as Button and ListView, from view models. By following the principles outlined here, view models can be tested in isolation, therefore reducing the likelihood of software defects by limiting scope.
+
+The following sections discuss the main approaches to connecting view models to views.
+
+## Creating a view model declaratively
+
+The simplest approach is for the view to declaratively instantiate its corresponding view model in XAML. When the view is constructed, the corresponding view model object will also be constructed. This approach is demonstrated in the following code example:
+
+```xaml
+<ContentPage ... xmlns:local="clr-namespace:eShop">
+    <ContentPage.BindingContext>
+        <local:LoginViewModel />
+    </ContentPage.BindingContext>
+    ...
+</ContentPage>
+```
+
+When the ContentPage is created, an instance of the LoginViewModel is automatically constructed and set as the view's BindingContext.
+
+This declarative construction and assignment of the view model by the view has the advantage that it's simple, but has the disadvantage that it requires a default (parameter-less) constructor in the view model.
+
+## Creating a view model programmatically
+
+A view can have code in the code-behind file, resulting in the view model being assigned to its BindingContext property. This is often accomplished in the view's constructor, as shown in the following code example:
+
+```csharp
+public LoginView()
+{
+    InitializeComponent();
+    BindingContext = new LoginViewModel(navigationService);
+}
+```
+
+The programmatic construction and assignment of the view model within the view's code-behind has the advantage that it's simple. However, the main disadvantage of this approach is that the view needs to provide the view model with any required dependencies. Using a dependency injection container can help to maintain loose coupling between the view and view model. For more information, see [Dependency injection](dependency-injection.md).
+
+## Updating views in response to changes in the underlying view model or model
+
+All view model and model classes that are accessible to a view should implement the [INotifyPropertyChanged](/dotnet/api/system.componentmodel.inotifypropertychanged) interface. Implementing this interface in a view model or model class allows the class to provide change notifications to any data-bound controls in the view when the underlying property value changes.
```

Comment:
```suggestion
All view model and model classes that are accessible to a view should implement the [<xref:System.ComponentModel.INotifyPropertyChanged> interface. Implementing this interface in a view model or model class allows the class to provide change notifications to any data-bound controls in the view when the underlying property value changes.
```

Are you using the [Docs Authoring Pack](https://docs.microsoft.com/contribute/how-to-write-docs-auth-pack) VS Code extension? If not, you're missing out on a lot of powerful features that help with writing for docs.microsoft.com. In particular there is a feature to add xref links: https://docs.microsoft.com/contribute/how-to-write-links#xref-cross-reference-links

Reply:
I am. This was the suggestion that it provided when adding a link. If a link points back to the docs site, should I always use xref? It has been wanting to use links like the above. I will look into this more now that I know!